Modules/Filtering/ImageCompare/test/itkImageFileReaderIOToRequestedRegionMismatchTest.cxx

```diff
@@ -0,0 +1,58 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         https://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+#include "itkSTAPLEImageFilter.h"
+#include "itkImageFileReader.h"
+#include "itkTestingMacros.h"
+
+
+int
+itkImageFileReaderIOToRequestedRegionMismatchTest(int argc, char * argv[])
+{
+  if (argc < 3)
+  {
+    std::cerr << "Missing parameters." << std::endl;
+    std::cerr << "Usage: " << itkNameOfTestExecutableMacro(argv) << "file1 file2 ... fileN" << std::endl;
+    return EXIT_FAILURE;
+  }
+
+  constexpr unsigned int Dimension = 2;
+
+  using InputImageType = itk::Image<unsigned short, Dimension>;
+  using OutputImageType = itk::Image<double, Dimension>;
+
+  using STAPLEImageFilterType = itk::STAPLEImageFilter<InputImageType, OutputImageType>;
+  auto stapleImageFilter = STAPLEImageFilterType::New();
+
+  typename itk::ImageFileReader<InputImageType>::Pointer reader;
+  for (int i = 1; i < argc; ++i)
+  {
+    // Instantiate a new reader for each image
+    reader = itk::ImageFileReader<InputImageType>::New();
```

Comment:
Not sure why instantiating a new reader is necessary, but the exception is not raised otherwise.

Reply:
If you set new inputs on the reader without calling update first, the output (filter's i-th input) will be an empty placeholder.

Reply:
Not sure if I get it. Calling [`reader->GetOutput()`](https://github.com/InsightSoftwareConsortium/ITK/pull/3611/files#diff-7c1978c198738c05c233bc960317a3ba88dfe2846aad5f0b47a5340ba7f6b2dcR49) should suffice to update the reader, right?

Reply:
Calling `reader->GetOutput()` does not update the reader. `Update()` needs to be called on it directly, or on some consuming filter downstream.